server/src/test/java/org/elasticsearch/cluster/metadata/DataStreamTests.java

```diff
@@ -125,7 +126,7 @@ protected DataStream mutateInstance(DataStream instance) {
                 ? null
                 : DataStreamLifecycle.newBuilder().dataRetention(randomMillisUpToYear9999()).build();
             case 10 -> {
-                failureIndices = randomValueOtherThan(List.of(), DataStreamTestHelper::randomIndexInstances);
+                failureIndices = randomIndexInstances();
```

Comment:
Shouldn't this be `randomNonEmptyIndexInstances()`? As the original line wants a value _other than_ an empty list.

Reply:
Nice catch! yes

Reply:
Ah actually not, I think this might have been a copy paste mistake because of the next lines:
```
if (failureIndices.isEmpty()) {
                    failureStore = false;
                } else {
                    failureStore = true;
                }
```
It can never be empty this way, so my guess here they intended to have either an empty one or a non empty list.

What do you think?

Reply:
Ah yeah you're right. But does it actually mutate something in both of those cases? I know it's not exactly related to this PR, but it looks like there's a chance nothing gets mutated (if the data stream didn't have a failure store and now still doesn't have one for instance).

Reply:
well it is mutating it because the chances to create exactly the same list is quite small. But let me ensure that's not the case, just in case.